File: Security/src/AspDotNet4/CloudFoundrySingleSignon-VB/App_Start/ApplicationConfig.vb
Imports System.IO
Imports Microsoft.Extensions.Configuration
Imports Microsoft.Extensions.Logging
Imports Steeltoe.Extensions.Configuration.CloudFoundry

Namespace CloudFoundrySingleSignon
    Public Class ApplicationConfig
        Public Shared Property Configuration As IConfiguration
        Public Shared Property LoggerFactory As LoggerFactory

        Public Shared Sub RegisterConfig(ByVal environment As String)
            Dim builder = New ConfigurationBuilder().SetBasePath(GetContentRoot()).AddJsonFile("appsettings.json", [optional]:=False, reloadOnChange:=False).AddEnvironmentVariables().AddCloudFoundry()
            Configuration = builder.Build()
            LoggerFactory = New LoggerFactory()
            LoggerFactory.AddConsole(LogLevel.Trace)
        End Sub

        Public Shared Function GetContentRoot() As String
            Dim basePath = If(CStr(AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY")), AppDomain.CurrentDomain.BaseDirectory)
            Return Path.GetFullPath(basePath)
        End Function
    End Class
End Namespace

File: Security/src/AspDotNet4/CloudFoundryWcf/ApplicationConfig.cs
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;
using Steeltoe.Extensions.Configuration.CloudFoundry;
using System;
using System.IO;

namespace CloudFoundryWcf
{
    public class ApplicationConfig
    {
        public static IConfiguration Configuration { get; set; }
        public static LoggerFactory LoggerFactory { get; private set; }

        public static void RegisterConfig(string environment)
        {
            // Set up configuration sources.
            var builder = new ConfigurationBuilder()
                .SetBasePath(GetContentRoot())
                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
                .AddJsonFile($"appsettings.{environment}.json", optional: true)
                .AddEnvironmentVariables()
                .AddCloudFoundry();

            Configuration = builder.Build();

            LoggerFactory = new LoggerFactory();
            LoggerFactory.AddConsole(LogLevel.Trace);
        }
        public static string GetContentRoot()
        {
            var basePath = (string)AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") ??
               AppDomain.CurrentDomain.BaseDirectory;
            return Path.GetFullPath(basePath);
        }
    }
}